### PR TITLE
feat(orderbook-api): order solution status

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Add `getOrderSolutionStatus` to orderbook api.

Based on types generated from openapi definition on https://github.com/cowprotocol/services/pull/2804

~TODO: update once backend merge/releases the PR~ merged, good to go!

https://barn.api.cow.fi/docs/#/default/get_api_v1_orders__UID__status

### CODE REVIEW

**Comment by alfetopito** [comment]
Please post merge review

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "prepare": "npm run build",
     "prepublishOnly": "npm test && npm run lint",
     "graphql:codegen": "graphql-codegen --config graphql-codegen.yml",
-    "swagger:codegen": " openapi --input https://raw.githubusercontent.com/cowprotocol/services/varbitrum/crates/orderbook/openapi.yml --output src/order-book/generated --exportServices false --exportCore false",
+    "swagger:codegen": " openapi --input https://raw.githubusercontent.com/cowprotocol/services/d8964a5751e8d2077c2db1502c2b429f5c6b50a4/crates/orderbook/openapi.yml --output src/order-book/generated --exportServices false --exportCore false",
```
Comment (anxolin) [inline package.json:30]: Would it make sense to use tagged versions instead? 

image

Comment (alfetopito) [reply package.json:30]: Yes, once they do tag it

Comment (anxolin) [reply package.json:30]: We are on the bleeding edge 🙃

If they deployed to prod, wouldn't this be tagged already?

Comment (alfetopito) [reply package.json:30]: It has been deployed to prod. However, that version had the issue with open api, which has not been released yet.
And we can't wait for them to deploy this to prod before cutting a version to use on CoW Swap during development.
Anyway, this time I published the RC.0 version and I have this PR pending (https://github.com/cowprotocol/cow-sdk/pull/217) until the version is published to prod.

Comment (anxolin) [reply package.json:30]: okok 👍

**Pull request merged**